### DISCUSSION TITLE
IModel does not contain a CreateBasicProperties method

### DISCUSSION THREAD

**Discussion by boskjoett** [rabbitmq/rabbitmq-dotnet-client#1254] [Q&A] [answered]
I am trying to use the RabbitMQ.Client library source code directly (not via NuGet package) in my project.
When I compile the project I get this error:
_'IModel' does not contain a definition for 'CreateBasicProperties' and no accessible extension method 'CreateBasicProperties' accepting a first argument of type 'IModel' could be found (are you missing a using directive or an assembly reference?)_

Why does this IModel implementation
[https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client/client/api/IModel.cs](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client/client/api/IModel.cs)
not contain a CreateBasicProperties method?

According to this documentation page it should
[https://rabbitmq.github.io/rabbitmq-dotnet-client/api/RabbitMQ.Client.IModel.html#RabbitMQ_Client_IModel_CreateBasicProperties](https://rabbitmq.github.io/rabbitmq-dotnet-client/api/RabbitMQ.Client.IModel.html#RabbitMQ_Client_IModel_CreateBasicProperties)

**Comment by boskjoett** [comment]
I can see that it is only missing on main branch.
It is defined in version 6.4.0
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/v6.4.0/projects/RabbitMQ.Client/client/api/IModel.cs
so I guess it has been obsoleted.

**Comment by boskjoett** [comment, marked as answer]
I tried to compile the library source code tagged 6.4.0, but it failed because this project file
[https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/v6.4.0/projects/RabbitMQ.Client/RabbitMQ.Client.csproj](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/v6.4.0/projects/RabbitMQ.Client/RabbitMQ.Client.csproj)
references this source code file that cannot be found
`<Compile Include="..\..\gensrc\autogenerated-api-0-9-1.cs" />`

How can I get the autogenerated-api-0-9-1.cs file?

**Reply by lukebakken** [reply]
Run `.\build.bat` first.
